### ISSUE TITLE
Text in 'send file' and avatar chooser dialogue are square boxes

### ISSUE THREAD

**Issue by Larry456** [qTox/qTox#4808]
##### Brief Description

OS: Fedora 27
qTox version: Nightly (compiled just now) - also affected last several versions I've compiled over the past 4 months
Commit hash:  It just says 'build without git' here
toxcore:  0.1.10 (compiled)
Qt:  5.9.2
…

Reproducible: Always

##### Steps to reproduce

Open the send file dialog box or try to choose a new avatar

##### Observed Behavior

All of the text is messed up

[Here is a screenshot](https://i.imgur.com/hTLHCHy.png)

----

Also there is nothing useful in the qtox log, nothing related to fonts when starting qtox or opening the dialog box, just random warnings about timers from hours/days ago

**Comment by sudden6** [comment]
Seems like you are missing some font needed by Qt. I don't think this is a problem with qTox.

**Comment by Larry456** [comment]
>Seems like you are missing some font needed by Qt. I don't think this is a problem with qTox.

This only affects qTox (I don't have any issues on other Qt programs), and is a recent issue (as in the last few months, I never used to have this issue with the program). It seems weird that this would only be affecting the dialogs of qTox (do they use a different font than the main window?) and how it wouldn't fall back to a different, default font. 

I know I haven't _uninstalled_ any fonts as I haven't had a reason to, I have **installed** a total of 2 fonts (not sure if that was around the time the issue started or not - /Sansation_Bold.ttf, and /Futurot.ttf) which are the only 2 changes to have been made to the default Fedora XFCE installation.

[Here is a list](https://pastebin.com/FM34FCny) of fonts from the fc-list command

Edit: I noticed a single font included in the source files for qTox (qTox-master/res/font/DejaVuSans.ttf), is this the font that qTox uses? Because I definitely have that font installed and haven't made any changes to it (or the dejavu-fonts-common, dejavu-sans-fonts, and dejavu-sans-mono-fonts packages I have installed) 

If I run qtox from a terminal it shows the following error output when I open a dialog for the first time, but I have DejaVuSans font installed according to my packages and fc-list (and assuming the TTF file included in the source gets built into the binary somehow)

```
(qtox:30640): Pango-WARNING **: failed to create cairo scaled font, expect ugly output. the offending font is 'DejaVu Sans 9.9990234375'

(qtox:30640): Pango-WARNING **: font_face status is: file not found

(qtox:30640): Pango-WARNING **: scaled_font status is: file not found

(qtox:30640): Pango-WARNING **: failed to create cairo scaled font, expect ugly output. the offending font is 'DejaVu Sans 9.9990234375'

(qtox:30640): Pango-WARNING **: font_face status is: file not found

(qtox:30640): Pango-WARNING **: scaled_font status is: file not found

(qtox:30640): Pango-WARNING **: shaping failure, expect ugly output. shape-engine='PangoFcShapeEngine', font='DejaVu Sans 9.9990234375', text='The quick brown fox jumps over the lazy dog.'

```

All of the fonts in the /usr/share/fonts/dejavu folder have the same permissions as the fonts in all the other font folders as well, so that shouldn't be an issue

**Comment by Larry456** [comment]
Ok, If I run qtox from a terminal it shows the following error output, but I have DejaVuSans font installed according to my packages and fc-list (and assuming the TTF file included in the source gets built into the binary somehow)

(qtox:30640): Pango-WARNING **: failed to create cairo scaled font, expect ugly output. the offending font is 'DejaVu Sans 9.9990234375'

(qtox:30640): Pango-WARNING **: font_face status is: file not found

(qtox:30640): Pango-WARNING **: scaled_font status is: file not found

(qtox:30640): Pango-WARNING **: failed to create cairo scaled font, expect ugly output. the offending font is 'DejaVu Sans 9.9990234375'

(qtox:30640): Pango-WARNING **: font_face status is: file not found

(qtox:30640): Pango-WARNING **: scaled_font status is: file not found

(qtox:30640): Pango-WARNING **: shaping failure, expect ugly output. shape-engine='PangoFcShapeEngine', font='DejaVu Sans 9.9990234375', text='The quick brown fox jumps over the lazy dog.'

**Comment by Larry456** [comment]
I've also confirmed that the default and current 'chat' font (specified in the qTox settings) is Dejavu Sans 14 point, and there are no issues displaying any chat, only the dialog boxes. So not sure why it would show the above errors in relation to Dejavu Sans for only dialog boxes.. Could the "9.9990234375" or the way the font name is used (spacing/location) be significant?

**Comment by sudden6** [comment]
Yes, qTox ships the Dejavu Sans font for it's own UI. But the file picker you showed with the broken font is supplied by your system.

From the looks of the file dialog and the logs I can see that in your case some GTK file picker is used. I guess you're using GNOME as Desktop environment then? Are you using some special HiDPI setup or something? Is it only the file dialogs that are broken?

A few releases back we switched from the default Qt file picker to the system provided one for better integration I guess this unmasked the problem.

>  Could the "9.9990234375" or the way the font name is used (spacing/location) be significant?

I don't have any idea how selecting fonts works, but the number looks very suspicious indeed.

**Comment by Larry456** [comment]
>From the looks of the file dialog and the logs I can see that in your case some GTK file picker is used. I guess you're using GNOME as Desktop environment then? Are you using some special HiDPI setup or something? Is it only the file dialogs that are broken?

I'm using XFCE (4.12), no HiDPI. As far as I'm aware those are the only things affected that I can find.

[Here's a screenshot](https://s18.postimg.org/xyyrbnn0p/image.png) of a working file dialog from another application (GIMP) with the same layout/appearance next to the one from qTox (assuming both programs are using the default system font for the dialogs, they should both be using sans 10px)

Edit: If I change the default system font in XFCE's settings, other fonts, even other dejavu fonts like dejavu seriff work. The default system font (sans regular 10px) as well as dejavusans book 10px both cause the boxes to appear, regardless of the font size (eg, sans 14px doesn't fix the boxes, only changing to a seperate font)

Also, if I use default sans/dejavu sans 12px or 14px, the terminal output when opening the dialog with boxes doesn't say 11.999.. or 13.999 like I thought it would based on my previous comment, but 12 and 14 respectively. So it doesn't seem like it's due to some kind of size conversion bug

Is it possible there is a conflict or problem with the included Dejavu sans font? It seems as though the file dialogs aren't using the system default font and is trying to use the included dejavu sans font, since it says 'File not found' in regards to the font and only has the issue when the default system font is 'sans' or dejavu sans (both of which _should_ be the same as the included font). Otherwise why would the same dialog in other programs display normally?

**Comment by Diadlo** [comment]
Screenshot from the post:
![image](https://user-images.githubusercontent.com/6306608/33287909-582f160c-d3cb-11e7-829c-d5bc19026906.png)

**Comment by TarisMarh** [comment]
The problem is still actual. All of dejavu-* packages are installed.
